docs/docsite/rst/porting_guides/porting_guide_core_2.19.rst

```diff
@@ -407,6 +407,74 @@ The result of the corrected template remains a list:
     }
 
 
+Example - unintentional ``None`` result
+"""""""""""""""""""""""""""""""""""""""
+
+If a template evaluated to ``None``, it was implicitly converted to an empty string in previous versions of ansible-core.
+This can now result in the template evaluating to the *value* ``None``.
```

Comment:
It's worth pointing out that this no longer matters for the common case of passing the template result to a module argument that accepts a string, since `None` is now automatically converted to an empty string in that case.

See: https://github.com/ansible/ansible/pull/85652

Reply:
Ah, I didn't notice that change. That doesn't work for `type=list` though, where an empty string as a result of a template in 2.18 and before resulted in `[""]`, and now results in an error. But since `[""]` was seldom the expected value I guess this doesn't matter much...

Reply:
This looks better.

Your point about lists has me wondering if they should default to `[]` when given `None` for module arguments, since that's the closest thing to "no list" -- just like we now do for strings defaulting to "" when `None` is given.

Reply:
Defaulting to `[]` would be changing behavior (since depending on where `None` comes from, it used to be `[""]` before). That sounds like a can of worm that can cause more bad than good. At least for strings, `None` either came from a template (and used to be `""`) or was invalid before.